README.md

```diff
@@ -35,36 +35,57 @@ $ npm install @sentinel-hub/sentinelhub-js
 
 ## Layers
 
-The core data structure is `Layer`, which corresponds to a _layer_ as returned by OGC WMS GetCapabilities request. If the URL matches one of the Sentinel Hub service URLs, additional capabilities are unlocked.
-
-Basic (WMS-capable) `Layer` can be initialized like this:
+The core data structure is `Layer`, which corresponds to a _layer_ as returned by OGC WMS GetCapabilities request. Basic (WMS-capable) `Layer` can be initialized like this:
 
 ```javascript
   import { WmsLayer } from '@sentinel-hub/sentinelhub-js';
 
-  const layer = new WmsLayer('https://services.sentinel-hub.com/ogc/wms/<your-instance-id>', '<layer-id>', 'Title', 'Description');
+  const layer = new WmsLayer({
+    baseUrl: 'https://services.sentinel-hub.com/ogc/wms/<your-instance-id>',
+    layerId: '<layer-id>',
+  });
 ```
 
-Such layer would only allow WMS requests. However, `Layer` is also a superclass for multiple dataset-specific subclasses (like `S1GRDAWSEULayer` - Sentinel-1 GRD data on AWS eu-central-1 Sentinel Hub endpoint) which can be instantiated with their own specific parameters and thus unlock additional powers.
+Such layer would only allow WMS requests. However, `Layer` is also a superclass for multiple dataset-specific subclasses (like `S1GRDAWSEULayer` - Sentinel-1 GRD data on AWS eu-central-1 Sentinel Hub endpoint) which can be instantiated with their own specific parameters and thus have additional capabilities.
 
 When it comes to Sentinel Hub layers, there are four ways to determine their content:
 
-- by `layerId`: ID of the layer, as used by OGC WMS and SentinelHub Configurator
+- by `instanceId` and `layerId`: ID of the layer, as used by OGC WMS and SentinelHub Configurator
 - by `evalscript`: custom (javascript) code that will be executed by the service per each pixel and will calculate the (usually RGB/RGBA) values
 - by `evalscriptUrl`: the URL from which the evalscript can be downloaded from
-- by `dataProduct`: the structure which contains an ID of a pre-existing product
+- by `dataProduct`: the ID of a pre-existing data product
```

Comment:
I think it would be appropriate to explain the difference between these approaches. Namely, that in the first case, only WMS can be used for `GetMap` etc. and only Processing for the rest. I am aware you mention this later, but it would be less confusing for a reader if explained here already.

Reply:
I see what you mean, but - the explanation is not that simple. InstanceId + layerId can be used for either WMS or Processing API (when authenticated) if the dataset supports it (because only SH v3 datasets support Processing API). If used with Processing API, configuration (`evalscript` and possibly other layer parameters) is fetched from service. `dataProduct` can only be used with SH v3, while evalscripts V3 are supported only on SH v3 (afaik)... and there are probably many more nuances. In other words, I don't have a good idea on how to improve this text without making it really complex... :) But maybe we can open a ticket there and try to word this a bit better? I agree that many users will have no idea what these 4 things are...

Reply:
I see what you mean, but - the explanation is not that simple. InstanceId + layerId can be used for either WMS or Processing API (when authenticated) if the dataset supports it (because only SH v3 datasets support Processing API). If used with Processing API, configuration (`evalscript` and possibly other layer parameters) is fetched from service. `dataProduct` can only be used with SH v3, while evalscripts V3 are supported only on SH v3 (afaik)... and there are probably many more nuances. In other words, I don't have a good idea on how to improve this text without making it really complex... :) But maybe we can open a ticket there and try to word this a bit better? I agree that many users will have no idea what these 4 things are...

Reply:
Make sense. Resolving.